xrdsst/controllers/base.py

```diff
@@ -34,6 +35,14 @@
 
 class BaseController(Controller):
     _DEFAULT_CONFIG_FILE = "config/xrdsst.yml"
+    _RE_API_KEY = re.compile(r"""
```

Comment:
This would better fit into the util.py where there is already "RE_API_KEY_HEADER"

Reply:
Yes, this code location was considered first, however it was put into the BaseController and marked with Python's largely useless 'protected' convention as it is directly used in output sanitation and safer out of ``core/util``, and  other packages where eager development happens more often.

However, there is a simple alternative, getting rid of the ``core/util.py`` ``RE_API_KEY_HEADER`` entirely by requiring configuration API key definition to match the one returned by the security server, solely relying on the 'protected' field of ``BaseController``, as clearly even the presence of so similar regular expressions is ``code smell``.

AFAIU, so far you have been against this simple change, but from the above it is maybe clear that this is best way to proceed, so please confirm your agreement here.

Reply:
Can not get rid of core/util.py as there are other functions/methods inside used also in other places in addition to the BaseController. My suggestion is: either move "RE_API_KEY_HEADER" also to BaseController or move "_RE_API_KEY" to core/util.py. If these are in the same place, it is more clear.

Reply:
> Can not get rid of core/util.py as there are other functions/methods inside used also in other places in addition to the BaseController. My suggestion is: either move "RE_API_KEY_HEADER" also to BaseController or move "_RE_API_KEY" to core/util.py. If these are in the same place, it is more clear.

Here you have misread my quote above:

>> However, there is a simple alternative, getting rid of the `core/util.py` `RE_API_KEY_HEADER` entirely by requiring configuration API key definition to match the one returned by the security server, solely relying on the 'protected' field of `BaseController`, as clearly even the presence of so similar regular expressions is `code smell`.

Removal was strictly about field in ``util.py`` and not about ``util.py`` itself). Since you did not answer the question about simply eliminating different forms of API keys above, I will ask once again, with the code sample according to latest suggestion:

Do you indeed still see any reason to have code like:

```python
class BaseController(Controller):
    _RE_API_KEY = re.compile(r"""
        ([a-f0-9]{8}-
        [a-f0-9]{4}-)
        ([a-f0-9]{4})-  # UUID version + 3 hexdecimal digits, only part retained
        ([a-f0-9]{4}-  # Do no validate first character separately
        [a-f0-9]{12})
    """, re.VERBOSE | re.IGNORECASE)

    # Regex of X-Road security server header for API key.
    RE_API_KEY_HEADER = re.compile(r"""
        ^
        X-Road-apikey[ ]token=
        [a-f0-9]{8}-
        [a-f0-9]{4}-
        [a-f0-9]{4}-  # Do not fix UUID version
        [a-f0-9]{4}-  # Do no validate first character separately
        [a-f0-9]{12}
        $
    """, re.VERBOSE | re.IGNORECASE)
```

when it simple and clear to just have  /single/ regular expression by having configuration API key definition match the one returned by the security server?

Reply:
I suppose if your refacto the code so that api-key in the configuration is kept without the "X-Road-apikey token=" you could easily remove the "RE_API_KEY_HEADER".

Reply:
Singe RE retained, https://github.com/nordic-institute/X-Road-Security-Server-toolkit/pull/43/commits/5eb996d0ee34135c33119a724893ff214846fb83